components/service/glean/src/main/java/mozilla/components/service/glean/timing/TimingManager.kt

```diff
@@ -44,38 +54,44 @@ internal object TimingManager {
      * preserved.
      *
      * @param metricData The metric managing the timing. Used for error reporting.
-     * @param timerId The object to associate with this timing.
+     * @return a [GleanTimerId] representing the id to associate with this timing.
      */
-    public fun start(metricData: CommonMetricData, timerId: Any) {
+    fun start(metricData: CommonMetricData): GleanTimerId? {
         val startTime = getElapsedNanos()
 
+        var timerId: GleanTimerId?
+
         synchronized(this) {
+            timerId = timerIdCounter++
+
             val metricName = metricData.identifier
             uncommittedStartTimes[metricName]?.let { metricTimings ->
                 if (timerId in metricTimings) {
```

Comment:
This should not ever be true anymore.

Reply:
If we don't find a solution for using an opaque timer id type and the user just passed in some integer they made up, this *could* happen, I suppose.

Reply:
Yup, we can't really make sure this doesn't happen. Also, I'm afraid I need to duplicate this code to the timespan :(